### PR TITLE
Fix several DOCKER_HOST assumptions in integration_test.py

### PR DESCRIPTION
**Pull request by tianon**
See https://github.com/docker/docker/pull/10797#issuecomment-74591463.

### CODE REVIEW

File: tests/integration_test.py [hunk 1]
```diff
@@ -1444,7 +1444,7 @@ def test_resource_warnings(self):
 
 class TestRegressions(unittest.TestCase):
     def setUp(self):
-        self.client = docker.client.Client(timeout=5)
+        self.client = docker.Client(timeout=5, base_url=DEFAULT_BASE_URL)
```
Comment (tianon) [inline tests/integration_test.py]: Is there a particular reason this one has a custom `timeout` value?

Comment (shin-) [inline tests/integration_test.py]: Per https://github.com/tianon/docker-py/blob/fix-host-assumptions/tests/integration_test.py#L50, all of the timeouts are set to 5 - any longer probably indicates a problem with the test anyway. Did you have any particular issue with this, or just curiosity?

**Comment by shin-** [comment]
LGTM!

File: tests/integration_test.py (hunk 1 above)
Comment (tianon) [inline tests/integration_test.py]: Just curiosity! :)

**Pull request merged**